detection-rules/impersonation_vip_urgent_request.yml

```diff
@@ -31,7 +31,11 @@ source: |
       and not profile.by_sender().any_false_positives
     )
   )
-
+  // negate sharepoint notifications originating from within the org
+  and not (
+      sender.email.email in ('no-reply@sharepointonline.com')
+      and all(headers.reply_to, .email.domain.root_domain in $org_domains)
+  )
```

Comment:
`and not (
  sender.email.email in ('no-reply@sharepointonline.com')
  and length(headers.reply_to) > 0
  and all(headers.reply_to, .email.domain.root_domain in $org_domains)
)
`

I think you actually want the length check first right, because all is vacuous it will evaluate true, if reply_to is null.